solutions/usaco-941.mdx

```diff
@@ -11,11 +11,161 @@ author: Jesse Choe, Ryan Chou
 ## Video Solution
 <Youtube id="4YTijr0BbrA" />
 
+## Explanation
+
+First, it may help to think of an instance where we cannot form a proper evolutionary tree. This would be an instance such that no matter how we form the tree, it would be inevitable that some characteristic would evolve in two distinct places in the tree. It turns out that the minimal such bad example looks like this:
+
+$\texttt{population1: A } \\
+\texttt{population2: B } \\
+\texttt{population3: A B}$
+
+In other words, we have a population with just trait $\texttt{A}$, a population with just trait $\texttt{B}$, and a population with both. If we want to build a tree out of this input, we would need to split on either $\texttt{A}$ or $\texttt{B}$ at the root, but then the remaining two subtrees would both need to have an edge that adds the other characteristic. For example, if the root split into "$\texttt{A}$" and "$\texttt{not A}$" branches, then both branches would need to contain an edge that adds the $\texttt{B}$ trait.
+
+It will help to actually look at things from the viewpoint of the characteristics instead of from the viewpoint of the populations, so let's "transpose" the input above:
+
+$\texttt{A: population1 population3 } \\
+\texttt{B: population2 population3}$
+
+The fundamental problem here is that there are populations in $\texttt{A}$ only, populations in $\texttt{B}$ only, and populations in both $\texttt{A}$ and $\texttt{B}$. If we look at the Venn diagram for the sets A and B, the picture therefore looks like this:
+
+![Venn Diagram 1](http://www.usaco.org/current/data/fig2_evolution_bronze_open19.png)
```

Comment:
do you think it would be better to put the images in a folder instead of linking to the usaco site, just in case the image is moved or smth in the future?

Reply:
+1